src/server/models/cacheable_queries/user.js

```diff
@@ -212,11 +218,41 @@ const userLoggedIn = async (field, val) => {
   return user;
 };
 
+const getAndClearNotifications = async userId => {
```

Comment:
getAndClearNotifications and addNotification are the crucial part of the backend.
While unusual, for these calls, we delete while we read, to avoid any 'old' notifications.  Besides making sure it's the user themselves that are doing the retrieving, this should stop any stale notifications from repeating.